config/config.go

```diff
@@ -574,30 +594,14 @@ func (c *Config) Parse() error {
 		"kubernetes-east-west-domain",
 	)
 
-	logLevel, err := log.ParseLevel(c.ApplicationLogLevelString)
-	if err != nil {
+	if err := Validate(c); err != nil {
 		return err
 	}
 
-	kubernetesPathMode, err := kubernetes.ParsePathMode(c.KubernetesPathModeString)
-	if err != nil {
-		return err
-	}
-
-	kubernetesEastWestRangePredicates, err := eskip.ParsePredicates(c.KubernetesEastWestRangePredicatesString)
-	if err != nil {
-		return fmt.Errorf("invalid east-west-range-predicates: %w", err)
-	}
-
-	histogramBuckets, err := c.parseHistogramBuckets()
-	if err != nil {
-		return err
-	}
-
-	c.ApplicationLogLevel = logLevel
-	c.KubernetesPathMode = kubernetesPathMode
-	c.KubernetesEastWestRangePredicates = kubernetesEastWestRangePredicates
-	c.HistogramMetricBuckets = histogramBuckets
+	c.ApplicationLogLevel, _ = log.ParseLevel(c.ApplicationLogLevelString)
+	c.KubernetesPathMode, _ = kubernetes.ParsePathMode(c.KubernetesPathModeString)
+	c.KubernetesEastWestRangePredicates, _ = eskip.ParsePredicates(c.KubernetesEastWestRangePredicatesString)
+	c.HistogramMetricBuckets, _ = c.parseHistogramBuckets()
```

Comment:
I think it was fine, maybe just move assignments (e.g. `c.ApplicationLogLevel = logLevel`) right after parsing error checks. This way we parse stuff twice but I guess its not a big deal.

Reply:
The problem is that we can not check the validations because of `flag.Parse` and this is only happening once at startup so I think it's fine to do this twice.

Reply:
Hm, I see not that you test `Validate`. Do we need it to be exported? There is also a chicken-and-egg problem here - we need to parse flags to get values we validate.

The global `flag.Parse()` is really unfortunate, maybe we could fix it using custom `flag.FlagSet` that we can stuff for testing, see https://eli.thegreenplace.net/2020/testing-flag-parsing-in-go-programs/

I did check if we can keep it backwards compatible for lib users.

Reply:
Validate->validate